infrastructures/docker/infrastructure/src/main/java/org/eclipse/che/workspace/infrastructure/docker/DockerRuntimeInfrastructure.java

```diff
@@ -52,70 +45,68 @@
 
   @Inject
   public DockerRuntimeInfrastructure(
-      EnvironmentParser dockerEnvironmentParser,
-      EnvironmentValidator dockerEnvironmentValidator,
       ContainersStartStrategy startStrategy,
       InfrastructureProvisioner infrastructureProvisioner,
       EnvironmentNormalizer environmentNormalizer,
       Map<String, DockerConfigSourceSpecificEnvironmentParser> environmentParsers,
       DockerRuntimeContextFactory contextFactory,
       DockerContainers containers,
-      EventService eventService,
-      InstallerRegistry installerRegistry,
-      RecipeRetriever recipeRetriever,
-      Set<InternalEnvironmentProvisioner> internalEnvironmentProvisioners,
-      ServersNormalizer serversNormalizer) {
-    super(
-        "docker",
-        environmentParsers.keySet(),
-        eventService,
-        installerRegistry,
-        recipeRetriever,
-        internalEnvironmentProvisioners,
-        serversNormalizer);
-    this.dockerEnvironmentValidator = dockerEnvironmentValidator;
-    this.dockerEnvironmentParser = dockerEnvironmentParser;
+      EventService eventService) {
+    super("docker", environmentParsers.keySet(), eventService);
```

Comment:
I don't think that `environmentParsers.keySet()` should be here. Actually, it supports only `dockerfile`, `dockerimage`, `compose`:
```
 switch (type) {
      case "dockerfile":
          ...
      case "dockerimage":
         ...
      case "compose":
        ...
      default:
        throw new InfrastructureException("Recipe type is not allowed " + type);
```
It's better to use ImmutableMap of supported recipes types